File: snippets/rust/src/receive_payment.rs
use std::sync::Arc;

use anyhow::Result;
use breez_sdk_core::*;

async fn receive_payment(sdk: Arc<BreezServices>) -> Result<()> {
    // ANCHOR: receive-payment
    let receive_payment_response = sdk
        .receive_payment(ReceivePaymentRequest {
            amount_msat: 3_000_000,
            description: "Invoice for 3000 sats".into(),
            ..Default::default()
        })
        .await?;

    let invoice = receive_payment_response.ln_invoice;
    // ANCHOR_END: receive-payment

    Ok(())
}

File: snippets/csharp/ReceivePayment.cs
using Breez.Sdk;

public class ReceivePaymentSnippets
{
    public void ReceivePayment(BlockingBreezServices sdk)
    {
        // ANCHOR: receive-payment
        try
        {
            var receivePaymentResponse = sdk.ReceivePayment(
                new ReceivePaymentRequest(3_000_000, "Invoice for 3000 sats"));

            var invoice = receivePaymentResponse.lnInvoice;
        }
        catch (Exception)
        {
            // Handle error
        }
        // ANCHOR_END: receive-payment
    }
}
